### PR TITLE
adds more configuration flags

### PR DESCRIPTION
**Pull request by SweeXordious**
Closes https://github.com/celestiaorg/celestia-app/issues/307 and https://github.com/celestiaorg/celestia-app/issues/506

### CODE REVIEW

File: e2e/qgb/deployer_test.go [hunk 1]
```diff
@@ -38,9 +38,9 @@ func TestDeployer(t *testing.T) {
 	HandleNetworkError(t, network, err, false)
 
 	// FIXME should we use the evm client here or go for raw queries?
-	evmClient := orchestrator.NewEvmClient(nil, bridge, nil, network.EVMRPC)
+	evmClient := orchestrator.NewEvmClient(nil, bridge, nil, network.EVMRPC, orchestrator.DEFAULTEVMGASLIMIT)
```
Comment (evan-forbes) [inline e2e/qgb/deployer_test.go:41]: should we be using the configured gas limit here, not the default?

File: e2e/qgb/deployer_test.go [hunk 2]
```diff
@@ -38,9 +38,9 @@ func TestDeployer(t *testing.T) {
...
 
 	eventNonce, err := evmClient.StateLastEventNonce(&bind.CallOpts{Context: network.Context})
 	assert.NoError(t, err)
-	assert.Equal(t, uint64(0), eventNonce)
+	assert.Equal(t, uint64(1), eventNonce)
```
Comment (evan-forbes) [inline e2e/qgb/deployer_test.go:45]: we might want to document the reasoning behind this change somewhere

Comment (evan-forbes) [reply e2e/qgb/deployer_test.go:45]: nvm, just saw #558

File: e2e/qgb/deployer_test.go (hunk 1 above)
Comment (SweeXordious) [reply e2e/qgb/deployer_test.go:41]: this is for e2e, doesn't matter much

Comment (SweeXordious) [reply e2e/qgb/deployer_test.go:41]: this is for e2e, doesn't matter much

Comment (SweeXordious) [reply e2e/qgb/deployer_test.go:41]: this is for e2e, doesn't matter much

Comment (evan-forbes) [reply e2e/qgb/deployer_test.go:41]: lol whoops! okie dokie

**Review by evan-forbes — APPROVED** [review]
:+1: :+1:

File: e2e/qgb/deployer_test.go (hunk 2 above)
Comment (SweeXordious) [reply e2e/qgb/deployer_test.go:45]: here https://github.com/celestiaorg/celestia-app/issues/373#issuecomment-1193336317

**Pull request merged**